src/Source/VectorTilesSource.js

```diff
@@ -79,7 +79,7 @@ class VectorTilesSource extends TMSSource {
             const s = Object.keys(style.sources)[0];
             const os = style.sources[s];
 
-            style.layers.forEach((layer) => {
+            style.layers.forEach((layer, order) => {
```

Comment:
`style.layers` is already sort, later we will try to use it.

Reply:
Yes, but it is quicker to use an index instead of doing indexOf everytime in the parser.